app/controllers/concerns/breadcrumbable.rb

```diff
@@ -15,6 +15,7 @@ def origin_pages_for(trainee)
     session[session_key_for(trainee)] ||= []
   end
 
+  # current_page only works for GET requests
```

Comment:
Curious - what happens when you use on a non GET request? Does it error nicely? If not it could be a bit confusing for debugging, maybe we could raise our own dev-friendly error.

Reply:
This was a bit of a rabbit hole. But basically, if you do say `rails routes | grep personal_details` you get the following routes:

```ruby
               trainee_personal_details_confirm GET    /trainees/:trainee_id/personal-details/confirm(.:format)                                          trainees/confirm_details#show
                  edit_trainee_personal_details GET    /trainees/:trainee_id/personal-details/edit(.:format)                                             trainees/personal_details#edit
                       trainee_personal_details GET    /trainees/:trainee_id/personal-details(.:format)                                                  trainees/personal_details#show
                                                PATCH  /trainees/:trainee_id/personal-details(.:format)                                                  trainees/personal_details#update
                                                PUT    /trainees/:trainee_id/personal-details(.:format)                                                  trainees/personal_details#update
```

Notice that rails doesn't give you routes for `PATCH` and `PUT`. We think this is also reason why although `Rails.application.routes.router.recognize(request) { |route, _| route.name }` finds a matching route, the `name` attribute for `routes.first.last` is `nil`.

Reply:
Ahhh interesting! That makes sense, the output for that first column `rails routes` is probably exactly the `name` attribute. Thanks!